CVRP example notebook

leonlan opened the pull request:
Closes #91.

N-Wouda commented:
I'll have a look at this tomorrow!

N-Wouda commented:
I would do this myself, but your branch is in a forked repository. But:

- Could you merge the docs on master into this branch?
- Add the notebook to `docs/source/examples/`, and list it in `docs/source/index.rst` under the travelling salesman example?

After that, the documentation should build automatically with this new notebook, and I can just read the web version to review it :).

leonlan commented:
Done!

N-Wouda commented:
Should we respect the vehicle limit in the notebook? Does the BKS do so?

leonlan commented:
I addressed most comments, and here are replies to some of the comments:

> - [ ] The example mentions assuming unlimited vehicles, but the instance described has 12 vehicles. Maybe explicitly mention that we ignore the vehicle constraint?

> Should we respect the vehicle limit in the notebook? Does the BKS do so?

I removed the line 'and 12 vehicles' when describing the instance. The CVRP problem description mentions an unlimited fleet size. The BKS assumes unlimited fleet as well. IIRC the `k12` is metadata that is associated to the instance, probably some lower bound.

> - [ ] Do we need the `cost` property?

Yes, for plotting we use the cost attribute because the bks also uses that.

N-Wouda commented:
LGTM. Can you run each modified example notebook from the top with a clean kernel? That way both the execution counts of each notebook line up, and, more importantly, we are sure that they execute as written without any dependency on weird variables that no longer exist.

Notebooks are nice, but their non-linear execution is very confusing.

N-Wouda commented on pyproject.toml:
```diff
@@ -44,6 +44,7 @@ flake8 = "^4.0.1"
 
 # These are solely for work on the example notebooks, and not required for
 # the package itself.
+cvrplib = "^0.1.1"
```
This is fine by me, but can we pull the instance from local (that is, from inside the repo)? I don't want to force users to have access to the internet to work with examples, nor have a dependency on a website that may or may not exist five years down the road.
